src/components/PopupWithForm.js

```diff
@@ -20,14 +29,27 @@ export default class PopupWithForm extends Popup {
     return fromValues;
   }
 
-  setEventListners() {
-    super.setEventListners();
-    this._popupForm.addEventListener("submit", (e) => {
-      this._handleFormSubmit(this._getinputvalues());
+  setEventListeners() {
+    super.setEventListeners();
+    this._popupForm.addEventListener("submit", async (e) => {
+      e.preventDefault();
+      this.setLoading(true);
+      this._submitButton.disabled = true;
+      const formData = this._getinputvalues();
+      //setTimeout(() => {
+      await this._handleFormSubmit(formData);
+      this._submitButton.textContent = this._defaultButtonText;
```

Comment:
You need to **only** return the **default** button text in the `finally` block, so that the button could get the text regardless of whether the request succeeds or fails.